structure_tests/types/v2/structure.go

```diff
@@ -141,6 +143,88 @@ func (st *StructureTest) RunFileContentTests(t *testing.T) int {
 	return counter
 }
 
+func (st *StructureTest) RunMetadataTests(t *testing.T) int {
+	t.Run(st.MetadataTest.LogName(), func(t *testing.T) {
+		driver, err := st.NewDriver()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		defer driver.Destroy()
+		config, err := driver.GetConfig(t)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		for _, pair := range st.MetadataTest.Env {
+			if config.Env[pair.Key] == "" {
+				t.Errorf("variable %s not found in image env", pair.Key)
+			} else if config.Env[pair.Key] != pair.Value {
+				t.Errorf("env var %s value does not match expected value: %s", pair.Key, pair.Value)
+			}
+		}
+
+		if len(st.MetadataTest.Cmd) > 0 {
```

Comment:
Not sure I follow this if/else if construct.

It looks like you're essentially trying to compare two arrays. Would a simple length check, then element by element check inside a for loop work?

Reply:
See my comment above for a bit more context. There are three cases here:

1) User specifies valid cmd in config
2) User specifies explicitly empty cmd in config
3) User does not specify any cmd in the config

These three cases are each caught by one part of the outer loop, in that order. It's a bit more clear if you imagine an empty `else {}` block at the end: this is for case 3, but if we've gotten here then we're guaranteed that case passes (because both `user.Cmd` and `config.Cmd` are empty), so the test passes and we continue.